rotkehlchen/assets/asset.py

```diff
@@ -1129,3 +1148,96 @@ def from_identifier(
             )
         except DeserializationError:
             return None
+
+
+# Create a generic variable that can be 'CustomAsset', or any subclass such as stock, painting, etc.  # noqa: E501
+R = TypeVar('R', bound='CustomAsset')
+
+
+@dataclass(init=True, repr=True, eq=False, order=False, unsafe_hash=False, frozen=True)
+class CustomAsset(Asset):
+    """Class to represent assets added by users that are not crypto coins"""
+    custom_asset_type: str = field(init=False)
+    notes: Optional[str] = field(init=False)
+
+    @classmethod
+    def initialize(  # type: ignore  # figure out a way to make mypy happy
+            cls: Type[R],
+            identifier: str,
+            name: str,
+            custom_asset_type: str,
+            notes: Optional[str] = None,
+            swapped_for: Optional[Asset] = None,
+            started: Optional[Timestamp] = None,
+    ) -> R:
+        """Initialize a custom asset from fields"""
+        custom_asset = cls('whatever', direct_field_initialization=True)
+        object.__setattr__(custom_asset, 'identifier', identifier)
+        object.__setattr__(custom_asset, 'name', name)
+        object.__setattr__(custom_asset, 'symbol', None)
+        object.__setattr__(custom_asset, 'asset_type', AssetType.CUSTOM_ASSET)
+        object.__setattr__(custom_asset, 'custom_asset_type', custom_asset_type)
+        object.__setattr__(custom_asset, 'notes', notes)
+        object.__setattr__(custom_asset, 'started', started)
+        object.__setattr__(custom_asset, 'forked', None)
+        object.__setattr__(custom_asset, 'swapped_for', swapped_for)
+        object.__setattr__(custom_asset, 'cryptocompare', None)
+        object.__setattr__(custom_asset, 'coingecko', None)
```

Comment:
Only adding them for compatibility? We need to figure out a better way here. This is really ugly

Also forked and swapped_for should not be here.